src/framework/standard/create_help_command.rs

```diff
@@ -176,40 +183,40 @@ impl CreateHelpCommand {
     #[cfg_attr(feature = "cargo-clippy", allow(useless_if_let_seq))]
     pub(crate) fn finish(self) -> Arc<Help> {
         if self.0.striked_commands_tip == Some(String::new()) {
-            let mut strike_text = String::from("~~`Striked commands`~~ are unavailable because they");
-
-            let mut concat_with_comma = if self.0.lacking_permissions == HelpBehaviour::Strike {
-                let _ = write!(strike_text, " require permissions");
-                true
-            } else {
-                false
-            };
-
-            if self.0.lacking_role == HelpBehaviour::Strike {
-
-                if concat_with_comma {
-                    let _ = write!(strike_text, ", require a specific role");
-                } else {
-                    let _ = write!(strike_text, " require a specific role");
-                    concat_with_comma = true;
-                }
-            }
-
-            if self.0.wrong_channel == HelpBehaviour::Strike {
-
-                if concat_with_comma {
-                    let _ = write!(strike_text, " or are limited to DM/guilds");
-                } else {
-                    let _ = write!(strike_text, " are limited to DM/guilds");
+            let mut strike_text =
+                String::from("~~`Striked commands`~~ are unavailable because they");
+
+            let reasons = [
+                (self.0.lacking_permissions, "require permissions"),
+                (self.0.lacking_role, "require a specific role"),
+                (self.0.lacking_ownership, "require ownership"),
+                (self.0.wrong_channel, "are limited to DM/guilds"),
+            ];
+            {
+                let mut concat_with_comma = false;
+                let mut iter = reasons
+                    .iter()
+                    .filter(|r| r.0 == HelpBehaviour::Strike)
+                    .map(|r| r.1)
+                    .peekable();
+                while let Some(text) = iter.next() {
+                    if concat_with_comma {
+                        let sep = if let None = iter.peek() { "," } else { ", or" };
```

Comment:
If I set all to `Strike`, I get the following output:
_Striked commands are unavailable because they require permissions, or require a specific role, or require ownership, are limited to DM/guilds._

Which uses the `, or`-separator for every listing but the last one, I assume switching these two would fix it, considering what `peek()` returns.

Reply:
I'm dumb. Fixed.